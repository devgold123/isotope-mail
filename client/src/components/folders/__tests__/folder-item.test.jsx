import React from 'react';
import {shallow} from 'enzyme/build/index';
import FolderItem from '../folder-item';

describe('FolderItem component test suite', () => {
  describe('Snapshot render', () => {
    test('Defaults, Should render FolderItem with no menu and no actions', () => {
      // Given
      const props = {
        className: 'the-one-percent', graphic: 'Andy Warhol', label: 'Red Label',
        selected: true, unreadMessageCount: 13, newMessageCount: 37};

      // When
      const folderItem = shallow(<FolderItem {...props} />);

      // Then
      expect(folderItem).toMatchSnapshot();
    });
    test('Defaults with actions, Should render FolderItem with hidden menu and actions', () => {
      // Given
      const props = {
        className: 'the-one-percent', graphic: 'Andy Warhol', label: 'Red Label',
        selected: true, unreadMessageCount: 13, newMessageCount: 37, onRename: () => {}, onDelete: () => {}};

      // When
      const folderItem = shallow(<FolderItem {...props} />);

      // Then
      expect(folderItem).toMatchSnapshot();
    });
    test('Context Menu Visible, Should render FolderItem with menu', () => {
      // Given
      const props = {
        className: 'the-one-percent', graphic: 'Andy Warhol', label: 'Red Label',
        selected: true, unreadMessageCount: 13, newMessageCount: 37, onRename: () => {}, onDelete: () => {}};

      // When
      const folderItem = shallow(<FolderItem {...props} />);
      folderItem.setState({contextMenuVisible: true});

      // Then
      expect(folderItem).toMatchSnapshot();
    });
  });
  describe('Component events', () => {
    test('Generic mouse events, events fired', () => {
      // Given
      const onDrop = jest.fn();
      const onClick = jest.fn();
      const folderItem = shallow(<FolderItem label={''} selected={false} onDrop={onDrop} onClick={onClick}/>);

      // When
      folderItem.find('.listItem').simulate('drop', {preventDefault: () => {}});
      folderItem.find('.listItem').simulate('dragOver', {
        preventDefault: () => {}, dataTransfer: {types: ['application/hal+json', 'text/plain', 'application/json']}});
      folderItem.find('.listItem').simulate('click');

      // Then
      expect(onDrop).toHaveBeenCalledTimes(1);
      expect(onClick).toHaveBeenCalledTimes(1);
    });
    test('Menu clicked, should show menu', () => {
      // Given
      const props = {
        selected: true, onRename: () => {}, onDelete: () => {}};
      const folderItem = shallow(<FolderItem {...props} />);

      // When
      folderItem.find('.listItem .actions').find({children: 'more_vert'})
        .simulate('click', {preventDefault: () => {}, stopPropagation: () => {}});

      // Then
      expect(folderItem.state('contextMenuVisible')).toEqual(true);
    });
    test('Mouse Leave, menu visible, should hide menu', () => {
      // Given
      const props = {
        label: 'Menu will hide', selected: true, onRename: () => {}, onDelete: () => {}};
      const folderItem = shallow(<FolderItem {...props} />);
      folderItem.setState({contextMenuVisible: true});

      // When
      folderItem.find('.listItem').simulate('mouseLeave');

      // Then
      expect(folderItem.state('contextMenuVisible')).toEqual(false);
    });
  });
});
