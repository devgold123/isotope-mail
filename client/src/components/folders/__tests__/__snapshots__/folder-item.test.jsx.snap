// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderItem component test suite Snapshot render Context Menu Visible, Should render FolderItem with menu 1`] = `
<a
  className="the-one-percent mdc-list-item listItem
        mdc-list-item--selected
        "
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onMouseLeave={[Function]}
  title="Red Label (13)"
>
  <span
    className="material-icons mdc-list-item__graphic graphic"
  >
    Andy Warhol
  </span>
  <span
    className="mdc-list-item__primary-text primaryText
          hasNewMessages"
  >
    Red Label (13)
  </span>
  <span
    className="actions"
  >
    <span
      className="contextMenu visible"
    >
      <i
        className="material-icons"
        onClick={[Function]}
      >
        delete
      </i>
      <i
        className="material-icons"
        onClick={[Function]}
      >
        edit
      </i>
    </span>
  </span>
</a>
`;

exports[`FolderItem component test suite Snapshot render Defaults with actions, Should render FolderItem with hidden menu and actions 1`] = `
<a
  className="the-one-percent mdc-list-item listItem
        mdc-list-item--selected
        "
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onMouseLeave={[Function]}
  title="Red Label (13)"
>
  <span
    className="material-icons mdc-list-item__graphic graphic"
  >
    Andy Warhol
  </span>
  <span
    className="mdc-list-item__primary-text primaryText
          hasNewMessages"
  >
    Red Label (13)
  </span>
  <span
    className="actions"
  >
    <span
      className="contextMenu "
    >
      <i
        className="material-icons"
        onClick={[Function]}
      >
        delete
      </i>
      <i
        className="material-icons"
        onClick={[Function]}
      >
        edit
      </i>
    </span>
    <i
      className="material-icons"
      onClick={[Function]}
    >
      more_vert
    </i>
  </span>
</a>
`;

exports[`FolderItem component test suite Snapshot render Defaults, Should render FolderItem with no menu and no actions 1`] = `
<a
  className="the-one-percent mdc-list-item listItem
        mdc-list-item--selected
        "
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onMouseLeave={[Function]}
  title="Red Label (13)"
>
  <span
    className="material-icons mdc-list-item__graphic graphic"
  >
    Andy Warhol
  </span>
  <span
    className="mdc-list-item__primary-text primaryText
          hasNewMessages"
  >
    Red Label (13)
  </span>
  <span
    className="actions"
  >
    <span
      className="contextMenu "
    />
  </span>
</a>
`;
